release: 2.77 (#17411)
snapcore/snapd: NEWS.md

# New in snapd 2.77
* FDE: add reseal check after snapd refresh
* FDE: allow proceeding without hardware root of trust in secboot
* FDE: allow reprovision without factory reset
* FDE: make reprovision only seal
* FDE: add reprovision recovery key generation API
* Interfaces: kernel-sched-ext-control | add the kernel sched-ext control interface implementation
* Interfaces: allow reading of /proc/self/smaps_rollup
* Interfaces: allow Wine to execute files accessed via the Document Portal
* Interfaces: allow gtk css in subdirectories

> * Interfaces: allow connecting to system-wide docker on classic

Reviewer: ```suggestion
* Interfaces: docker | allow connecting to system-wide docker on classic
```

Generally when applicable it would be nice to prefix the interface name

Author: tx

* Interfaces: allow the systemd networkctl command
* Interfaces: allow systemd networkd link property changes via D-Bus
* Interfaces: grant default access to memory.high in a snap's cgroup
* Interfaces: fix mountinfo denial for steam and docker interfaces
* Interfaces: add xdg-portal-permission-store interface
* Interfaces: add atkey pid and 7 relative vid support
* Interfaces: make polkit and upower implicit on Core systems only
* Confdb: add validation-sets handler and fix data loss when writing to new schemas or accounts
* Confdb: fix bug on reading uneven lists
* Confdb: support sign-only external keypair backends
* Confdb: support Encode/Decode for builtins
* Remote device management: add post install actions API, apply messages, queue response messages, and sequencing for dispatch of management messages
* Packaging: assign a default label for /tmp/snap-private-tmp and set it during installation
* Packaging: fix service startup during install and session-agent socket handling on Ubuntu 26.04+
* Packaging: update bundled AppArmor to 5.0.2 and accept the 5.0 ABI when running as deb
* Packaging: use a relative symlink for snapctl and update steam-support udev rules

> * Packaging: add back gbp.conf output directory for Ubuntu 26.04 builds

Reviewer: I think we also need a `Packaging:` entry to explain the changes to go version used in the deb:

```suggestion
* Packaging: add back gbp.conf output directory for Ubuntu 26.04 builds
* Packaging: build deb with Go 1.23 for noble and jammy, Go 1.22 for focal
```

Author: added

* Cross-distro: update SELinux policy to allow snapd socket activation
* Cross-distro: update SELinux policy to support /etc/ld.so.cache mounting
* Snap: add run-inhibit hints, better component reporting, snap delta support, and improved install handling
* snap-confine: work around kernel mnt_ns_loop() ordering bug on 6.18.x
* Snap-confine: allow inheriting unix sockets from snaps, link to libm, and harden bpffs mounts
* Snap-confine: fix mountinfo parsing, set FD_CLOEXEC on BPF helper fds, and support transparent huge pages
* Snap-update-ns: switch to a multi-pass process for constructing and updating mount namespaces
* Core-initrd: add missing libbpf and systemd dlopen dependencies, and increase mount burst
* Security logging: add seclog API for administrative actions and token create/remove events
* Prompting: re-enable the prompting notice backend and improve validation and error handling

> * Cmd/snap: report hidden file access when prompting is active and fix self-managed cgroup support checks

Reviewer: Feels like these are two different things? The changes to `snap routine file-access` are around hidden file access when prompting is active, and there's another one related to `system-package-doc` as well (#17205). I think the cgroup support checks are unrelated? At least, I think the prompting and `system-package-doc` deserve their own item(s), and the cgroup support checks should be separate.

```suggestion
* Cmd/snap: report hidden file access for paths allowed by home when prompting is active
* Cmd/snap: report read-only file access for paths allowed by system-package-doc
* Cmd/snap: fix self-managed cgroup support checks
```

# New in snapd 2.77

> * FDE: add reseal check after snapd refresh

Reviewer 2: We need an entry for https://github.com/canonical/snapd/pull/17324, something like:
```
FDE: extend /v2/system-info/storage-encrypted to show install-time decisions
```

Author: added

> * FDE: allow proceeding without hardware root of trust in secboot

Reviewer 2: this already has an entry under 2.76.3 "FDE: bump secboot to rev 457b03a16d19"

* FDE: allow reprovision without factory reset
* FDE: make reprovision only seal

> * FDE: add reprovision recovery key generation API

Reviewer 3: what's the entry for https://github.com/canonical/snapd/commit/983808afb4fee99933bfb8b6c3cd3c1ab4718f22 ?

Author: This is the pr https://github.com/canonical/snapd/pull/17034

Reviewer 3: something is wrong becaue that PR/API is more thatn just the recovery key generation, we want something like

* FDE: add reprovision API

then
